data/pools/transactionPool.go

```diff
@@ -238,6 +242,7 @@ func (pool *TransactionPool) rememberCommit(flush bool) {
 
 	if flush {
 		pool.pendingTxGroups = pool.rememberedTxGroups
+		pool.stateproofOverflowed = false
```

Comment:
nit: maybe this line should go into `recomputeBlockEvaluator`

Reply:
I prefer leaving it as is, it is simpler with the locks. 
Moving it to recomputeBlockEvaluator needs to lock the `pendingMu`.